pydantic_ai_service.py

```
@@ -0,0 +1,107 @@
+
+            if not content:
+                return ErrorResponse(
+                    error="Empty response received from LLM",
+                    error_type="EmptyResponseError"
+                )
+
+            # Try to parse the response into the expected Pydantic model
+            try:
+                # If the response is JSON, parse it directly
+                parsed_response = response_model.model_validate_json(content)
+                return parsed_response
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)